src/js/tabs/configuration.js

```diff
@@ -516,47 +521,68 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
-            let rpmFeaturesVisible = $("input[id='dshotBidir']").is(':checked') || $("input[name='ESC_SENSOR']").is(':checked');
-            $('div.motorPoles').toggle(rpmFeaturesVisible);
-        }
-
         $('#escProtocolTooltip').toggle(semver.lt(CONFIG.apiVersion, "1.42.0"));
         $('#escProtocolTooltipNoDSHOT1200').toggle(semver.gte(CONFIG.apiVersion, "1.42.0"));
 
-        esc_protocol_e.val(PID_ADVANCED_CONFIG.fast_pwm_protocol + 1);
-        esc_protocol_e.change(function () {
-            var escProtocolValue = parseInt($(this).val()) - 1;
-
-            var newValue;
-            if (escProtocolValue !== PID_ADVANCED_CONFIG.fast_pwm_protocol) {
-                newValue = $(this).find('option:selected').text();
+        function updateVisibility() {
+            // Hide unused settings
+            const protocolName = $('select.escprotocol option:selected').text();
+            const protocolConfigured = protocolName !== 'DISABLED';
+            let digitalProtocol = false;
+            switch (protocolName) {
+            case 'DSHOT150':
+            case 'DSHOT300':
+            case 'DSHOT600':
+            case 'DSHOT1200':
+            case 'PROSHOT1000':
+                digitalProtocol = true;
+
+                break;
+            default:
+
+                break;
             }
-            self.analyticsChanges['EscProtocol'] = newValue;
 
-            //hide not used setting for DSHOT protocol
-            let digitalProtocol = (escProtocolValue >= self.DSHOT_PROTOCOL_MIN_VALUE);
+            let rpmFeaturesVisible = $("input[id='dshotBidir']").is(':checked') || $("input[name='ESC_SENSOR']").is(':checked');
+
+            $('div.minthrottle').toggle(protocolConfigured && !digitalProtocol);
+            $('div.maxthrottle').toggle(protocolConfigured && !digitalProtocol);
+            $('div.mincommand').toggle(protocolConfigured && !digitalProtocol);
+            $('div.checkboxPwm').toggle(protocolConfigured && !digitalProtocol);
+            $('div.unsyncedpwmfreq').toggle(protocolConfigured && !digitalProtocol);
 
-            $('div.minthrottle').toggle(!digitalProtocol);
-            $('div.maxthrottle').toggle(!digitalProtocol);
-            $('div.mincommand').toggle(!digitalProtocol);
-            $('div.checkboxPwm').toggle(!digitalProtocol);
-            $('div.unsyncedpwmfreq').toggle(!digitalProtocol);
+            $('div.digitalIdlePercent').toggle(protocolConfigured && digitalProtocol);
+            $('.escSensor').toggle(protocolConfigured && digitalProtocol);
 
-            $('div.digitalIdlePercent').toggle(digitalProtocol);
-            $('.escSensor').toggle(digitalProtocol);
+            $('div.checkboxDshotBidir').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0") && digitalProtocol);
+            $('div.motorPoles').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0"));
+            $('div.motorPoles').toggle(protocolConfigured && rpmFeaturesVisible);
```

Comment:
@mikeller We have problem hidding it with non digital protocol, I forgot it when created this motorPoles hidde code, can you change to this check?

```
            $('div.motorPoles').toggle(protocolConfigured && semver.gte(CONFIG.apiVersion, "1.42.0"));
            $('div.motorPoles').toggle(digitalProtocol && rpmFeaturesVisible);
```

Reply:
maybe we can move `protocolConfigured` and `digitalProtocol` into `rpmFeaturesVisible` and leave a single toggle line for msp version and `rpmFeaturesVisible` ??